zio-http/src/main/scala/zhttp/api/Handler.scala

```diff
@@ -0,0 +1,27 @@
+package zhttp.api
+
+import zhttp.http.HttpApp
+import zio.ZIO
+
+final case class Handler[-R, +E, Params, Input, Output](
+  api: API[Params, Input, Output],
+  handle: ((Params, Input)) => ZIO[R, E, Output],
+) { self =>
+
+  def ++[R1 <: R, E1 >: E](that: Handler[R1, E1, _, _, _]): Handlers[R1, E1] =
+    Handlers(this) ++ that
+
+  def toHttp: HttpApp[R, E] =
+    ServerInterpreter.handlerToHttpApp(self)
```

Comment:
```suggestion
  def toHttp: HttpApp[R, E] = self match {
    case Concat(left, right) => left.handle ++ right.handle
    case Single(_, handle)   => http
  }
```
- This way we won't need a `ServerInterpreter`
- We can also mount the `Handler` on any route as an HttpApp